RWKV-v4neo/src/model.py

```diff
@@ -151,21 +151,20 @@ def RUN_CUDA(B, T, C, w, u, k, v):
 
 
 class LoraLinear(nn.Module):
-
     def __init__(self, in_features: int, out_features: int, bias: bool):
         super().__init__()
 
-        self.weight = nn.Parameter(torch.empty((out_features, in_features)))
+        self.weight = nn.Parameter(torch.empty((out_features, in_features), dtype=DTYPE), requires_grad=False)
         assert bias == False, "Biased LoraLinear not supported"
 
         r, alpha, dropout = LORA_CONFIG["r"], LORA_CONFIG[
             "alpha"], LORA_CONFIG["dropout"]
-        self.lora_A = nn.Parameter(torch.empty(r, in_features))
-        self.lora_B = nn.Parameter(torch.empty(out_features, r))
+        self.lora_A = nn.Parameter(torch.empty(r, in_features, dtype=DTYPE))
+        self.lora_B = nn.Parameter(torch.empty(out_features, r, dtype=DTYPE))
         self.lora_dropout = nn.Dropout(dropout)
         self.scaling = alpha / r
 
-        nn.init.kaiming_uniform_(self.weight, a=math.sqrt(5))
+        nn.init.zeros_(self.weight)
```

Comment:
The initialization assumes the base model is uninitialized so a Kaiming init is applied as is in a usual linear layer. It's not really necessary though.